src/databricks/sql/utils.py

```diff
@@ -594,11 +615,15 @@ def named_parameters_to_tsparkparams(parameters: Union[List[Any], Dict[str, str]
         dbsql_params = named_parameters_to_dbsqlparams_v2(parameters)
     inferred_type_parameters = infer_types(dbsql_params)
     for param in inferred_type_parameters:
-        tspark_params.append(
-            TSparkParameter(
-                type=param.type.value,
-                name=param.name,
-                value=TSparkParameterValue(stringValue=param.value),
+        # The only way to pass a VOID/NULL to DBR is to declare TSparkParameter without declaring
+        # its value or type arguments. If we set these to NoneType, the request will fail with a
+        # thrift transport error
+        if param.type == DbSqlType.VOID:
```

Comment:
Recommend making a real type like DbsqlDynamicDecimalType, and then having each type implement a protocol for producing their TSparkParameter.  Then this part of the method becomes

...
[inferred_param.to_tspark_parameter() for inferred_param in inferred_type_parameters]

Reply:
This is really good feedback. Will push to a second pull request shortly.